### PR TITLE
Adding L431RC  Generic

### PR DESCRIPTION
**Pull request by mackelec**
Adding L431RC generic.
Added  ldscript.ld
Edited  Boards and Readme

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -4954,6 +4954,14 @@ GenL4.menu.pnum.GENERIC_L422KBUX.build.board=GENERIC_L422KBUX
...
 GenL4.menu.pnum.GENERIC_L422KBUX.build.variant=STM32L4xx/L412K(8-B)(T-U)_L422KB(T-U)
 
+# Generic L431RCTx
+GenL4.menu.pnum.GENERIC_L431RCTX=Generic L431RCTx
+GenL4.menu.pnum.GENERIC_L431RCTX.upload.maximum_size=262144
+GenL4.menu.pnum.GENERIC_L431RCTX.upload.maximum_data_size=65536
+GenL4.menu.pnum.GENERIC_L431RCTX.build.board=GENERIC_L431RCTX 
+GenL4.menu.pnum.GENERIC_L431RCTX.build.product_line=STM32L431xx
+GenL4.menu.pnum.GENERIC_L431RCTX.build.variant=STM32L4xx/L431R(B-C)(I-T-Y)
+
```
Comment (ABOSTM) [inline boards.txt:4964]: Would be great if you could add all the generic MCU you can found in boards_entry.txt, as they share the same configuration files (it is just a copy and past of boards_entry.txt, but don't forget to also add them in README.md)

File: variants/STM32L4xx/L431R(B-C)(I-T-Y)/ldscript.ld [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+_Min_Heap_Size = 0x200 ; /* required amount of heap */
+_Min_Stack_Size = 0x400 ; /* required amount of stack */
+
+/* Memories definition */
+MEMORY
+{
+  RAM    (xrw)    : ORIGIN = 0x20000000,   LENGTH = 64K
+  RAM2    (xrw)    : ORIGIN = 0x10000000,   LENGTH = 16K
+  FLASH    (rx)    : ORIGIN = 0x8000000,   LENGTH = 256K
```
Comment (ABOSTM) [inline variants/STM32L4xx/L431R(B-C)(I-T-Y)/ldscript.ld:50]: Could you please update .ld file according to wiki:
https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29#3---generic-system-clock-configuration

**Review by ABOSTM — CHANGES_REQUESTED** [review]
Thanks @mackelec for this Pull Request.
It would be great if you could also update the generic clock so that users can benefit from 80Mhz (thanks to PLL),  instead of relying on default 4Mhz clock (MSI).
https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29#3---generic-system-clock-configuration

**Comment by mackelec** [comment]
Thank you for your review.  I hope I've made all the correct updates this time.

**Review by ABOSTM — CHANGES_REQUESTED** [review]
@mackelec,
Thanks for this update,
It looks like you kind of revert several merged commit.
I did not list all but there is clearly too much update in board.txt

File: boards.txt [hunk 3]
```diff
@@ -2900,44 +2900,12 @@ GenF7.name=Generic STM32F7 series
...
 GenF7.build.core=arduino
 GenF7.build.board=GenF7
-GenF7.build.st_extra_flags=-D{build.product_line} {build.enable_usb} {build.xSerial} {build.bootloader_flags}
+GenF7.build.extra_flags=-D{build.product_line} {build.enable_usb} {build.xSerial} {build.bootloader_flags}
 GenF7.build.mcu=cortex-m7
 GenF7.build.flags.fp=-mfpu=fpv4-sp-d16 -mfloat-abi=hard
 GenF7.build.series=STM32F7xx
 GenF7.build.cmsis_lib_gcc=arm_cortexM7lfsp_math
 
-# Generic F722RCTx
```
Comment (ABOSTM) [inline boards.txt:2909]: Looks like revert of PR #1722

**Review by ABOSTM — CHANGES_REQUESTED** [review]

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -407,7 +407,8 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
 | :green_heart:  | STM32L412K8<br>STM32L412KB<br>STM32L422KB | Generic Board | *2.0.0* |  |
-| :green_heart:  | STM32L432KB<br>STM32L432KC<br>STM32L442KC | Generic Board | *2.0.0* |  |
+| :green_heart:  | STM32L412K8<br>STM32L412KB<br>STM32L422KB | Generic Board | *2.0.0* |  |
```
[the remark this answers is not in the document]
Comment (mackelec) [reply README.md:410]: I'm sorry Alexandre, I'll look at it tomorrow,

File: README.md [hunk 5]
```diff
@@ -407,6 +407,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
 | :green_heart:  | STM32L412K8<br>STM32L412KB<br>STM32L422KB | Generic Board | *2.0.0* |  |
+| :green_heart:  | STM32L431RB<br>STM32L431RC | Generic Board | **2.3.0** |  |
```
Comment (fpistm) [inline README.md:410]:
```suggestion
| :yellow_heart:  | STM32L431RB<br>STM32L431RC | Generic Board | **2.3.0** |  |
```

File: variants/STM32L4xx/L431R(B-C)(I-T-Y)/ldscript.ld [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+******************************************************************************
+**
+** @file        : LinkerScript.ld
+**
+** @author      : Auto-generated by STM32CubeIDE
+**
+** @brief       : Linker script for STM32L431RCTx Device from STM32L4 series
+**                      256Kbytes FLASH
+**                      64Kbytes RAM
+**                      16Kbytes RAM2
```
Comment (fpistm) [inline variants/STM32L4xx/L431R(B-C)(I-T-Y)/ldscript.ld:11]: This seems wrong based on datasheet:

> Embedded SRAM
> STM32L431xx devices feature 64 Kbyte of embedded SRAM. This SRAM is split into two
> blocks:
> • 48 Kbyte mapped at address 0x2000 0000 (SRAM1)
> • 16 Kbyte located at address 0x1000 0000 with hardware parity check (SRAM2).
> This memory is also mapped at address 0x2000 C000, offering a contiguous address
> space with the SRAM1 (16 Kbyte aliased by bit band)
> This block is accessed through the ICode/DCode buses for maximum performance.
> These 16 Kbyte SRAM can also be retained in Standby mode.
> The SRAM2 can be write-protected with 1 Kbyte granularity.
> The memory can be accessed in read/write at CPU clock speed with 0 wait states.

File: variants/STM32L4xx/L431R(B-C)(I-T-Y)/ldscript.ld [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+_estack = ORIGIN(RAM) + LENGTH(RAM); /* end of "RAM" Ram type memory */
+
+_Min_Heap_Size = 0x200 ; /* required amount of heap */
+_Min_Stack_Size = 0x400 ; /* required amount of stack */
+
+/* Memories definition */
+MEMORY
+{
+  RAM    (xrw)    : ORIGIN = 0x20000000,   LENGTH = LD_MAX_DATA_SIZE
+  RAM2    (xrw)    : ORIGIN = 0x10000000,   LENGTH = 16K
```
Comment (fpistm) [inline variants/STM32L4xx/L431R(B-C)(I-T-Y)/ldscript.ld:49]: Ditto,
as boards.txt specify 64 k,  RAM2 should not be define or RAM should be 48k

File: boards.txt [hunk 8]
```diff
@@ -7967,4 +8015,4 @@ Midatronics.menu.rtlib.nanofs.build.flags.ldspecs=--specs=nano.specs -u _scanf_f
 Midatronics.menu.rtlib.nanofps=Newlib Nano + Float Printf/Scanf
 Midatronics.menu.rtlib.nanofps.build.flags.ldspecs=--specs=nano.specs -u _printf_float -u _scanf_float
 Midatronics.menu.rtlib.full=Newlib Standard
-Midatronics.menu.rtlib.full.build.flags.ldspecs=
+Midatronics.menu.rtlib.full.build.flags.ldspecs=
```
Comment (fpistm) [inline boards.txt:8018]: Don't know why this line is changed?

**Review by fpistm — CHANGES_REQUESTED** [review]

**Review by fpistm — APPROVED** [review]
LGTM. Thanks @mackelec & @ABOSTM

**Pull request merged**